sakuma/ts/src/Todo.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from 'react';
+import initialList, {listType} from'./initialList';
+import TodoTabPanel from'./TodoTabPanel';
+import AddTodo, { onAddTodoProps } from'./AddTodo';
+import { Tabs, Tab, Box } from "@mui/material";
+import { styled } from '@mui/system';
+
+const TAB_NAME = {
+    all: 'すべて',
+    over: '過ぎているタスク',
+    completed: '完了',
+};
+
+const TodoComponent = styled('div')`
+    max-width: 400px;
+    margin:100px auto;
+    display: flex;
+    flex-wrap: wrap;
+`;
+
+const TodoContainer = styled('div')`
+    width:100%;
+`;
+
+type tabListType = {
+    [key: string] : listType[];
+};
```

Comment:
このままだとキーが分からないですね。

- 今ある三つ以外のどのキーも使えるようになるのはよくないです
- VSCodeなどの補助機能は発動しません


```suggestion
type tabListType = {
   all: listType[];
   completed: listType[];
   over: listType[];
};
```

※実際にactiveTabも同じく、TAB_NAMEにないものが設定できなくする、などがいいと思ってます。

もしTypescriptに怒られたら、TabのComponentをマップせず、全部ベタ書きでもいいです

```tsx
        <Tabs >
            <Tab key="all" value="all" label={TAB_NAME.all+'('+tabList.all.length+')'} />
            <Tab key="completed" value="completed" label={TAB_NAME.completed+'('+tabList.completed.length+')'} />
            <Tab key="over" value="ober" label={TAB_NAME.over+'('+tabList.over.length+')'} />
        </Tabs>
```

Reply:
Typescriptは具体的にかけるものを具体的に書かせる傾向があるので、面倒くさいと思うところもあるかもしれませんが、固定で数個あるもの（今回のall/completed/over）は出来るだけmapせず具体的に書いた方がいいと思います。

Reply:
mapによる`tabList[tabName]`という書き方が`key: string`と指定しないとエラーになるためこのような書き方をしていました。
色々調べたのですが、この場合は`const tabList`でkeyの種類（all, completed, over）を指定することにより、tabListで他のkeyを指定できないようにさせるという考え方みたいです。
```
type TabList = {
    [key: string] :List[];
};
const tabList: TabList = {
        all: ...,
        completed: ...,
        over: ...
    };
```
確かに`type TabList`自体はあらゆるkeyを指定できてしまいますが、`tabList`の方で制限する、という考え方で使用するみたいですね。

ただ、具体的に書くという慣習があるのであればそうした方が良いと思うので、指摘いただいた通りに修正します！